src/widget.js

```diff
@@ -40,32 +40,39 @@ var widget = {
             var parentNode = widgets[i];
             var type = parentNode.dataset.type;
             var widget_name =  type + "_widget" + i;
-            
             var keyword = '';
+
+            /* 
+                If key is search or all, update both widgets, otherwise update one and skip the other since this
+                iterates all widgets every time a click event is fired. Not continuing would cause both widgets to
+                reload on any click, creating a bad user experience. 
+            */
+
```

Comment:
Good! I was about to add a comment that we will generally need more code comments in your PR :) :+1: